### ISSUE TITLE
Typescript's standard fixer does not support "Optional chaining"  syntax

### ISSUE THREAD

**Issue by dangtrongnghiamq** [dense-analysis/ale#3203]
## Information

**VIM version**

NVIM v0.4.3
Build type: Release

Operating System: Window 10 Pro version 1903

## What went wrong

When i updated the new version. typescript's standard fixer does not support "Optional chaining" syntax. 

![image](https://user-images.githubusercontent.com/47862494/84342060-d0a04b00-abce-11ea-9af6-15302d038fc8.png)

I tried deleting and reinstalling but it failed.

## Reproducing the bug

1. My setting .vimsrc

let g:ale_sign_error = '>-'
let g:ale_sign_warning = '--'
let g:ale_linters = {
\   'javascript': ['eslint'],
\   'typescript': ['tslint'],
\   'python': ['pylint'],
\}

let g:ale_fixers = {
\   '*': ['remove_trailing_lines', 'trim_whitespace'],
\   'javascript': ['eslint'],
\   'typescript': ['tslint'],
\   'python': ['black']
\}

let g:ale_fix_on_save = 1
let g:ale_completion_enabled = 1

2. Setting package.json
` "devDependencies": {
    "@babel/cli": "^7.8.4",
    "@babel/core": "^7.9.0",
    "@babel/helper-call-delegate": "^7.8.7",
    "@babel/node": "^7.8.7",
    "@babel/plugin-proposal-class-properties": "^7.8.3",
    "@babel/preset-env": "^7.9.0",
    "@babel/preset-react": "^7.9.4",
    "@babel/register": "^7.9.0",
    "@types/enquire.js": "^2.1.2",
    "@types/lodash": "^4.14.149",
    "@types/node": "^13.7.7",
    "@types/react": "^16.9.23",
    "@types/react-redux": "^7.1.7",
    "babel-loader": "^8.1.0",
    "glob": "^7.1.6",
    "nodemon": "^2.0.2",
    "typescript": "^3.8.3",
    "webpack-cli": "^3.3.11",
    "webpack-glob-entry": "^2.1.1",
    "webpack-node-externals": "^1.7.2",
    "webpackbar": "^4.0.0"
  },
`

### :ALEInfo

 Current Filetype: json
Available Linters: ['jsonlint']
  Enabled Linters: ['jsonlint']
 Suggested Fixers: 
  'fixjson' - Fix JSON files with fixjson.
  'jq' - Fix JSON files with jq.
  'prettier' - Apply prettier to a file.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

let g:ale_json_jsonlint_executable = 'jsonlint'
let g:ale_json_jsonlint_use_global = 0
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = 100
let g:ale_completion_enabled = 1
let g:ale_completion_max_suggestions = 50
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 1
let g:ale_fixers = {'typescript': ['tslint'], '*': ['remove_trailing_lines', 'trim_whitespace'], 'javascript': ['eslint'], 'python': ['black']}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'typescript': ['tslint'], 'javascript': ['eslint'], 'python': ['pylint']}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>-'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>-'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - failure) jsonlint

**Comment by marcagba** [comment]
Hello @dangtrongnghiamq !
I found you issue while searching about a similar error. Eventually my error was caused by an outdated typescript version.

> When i updated the new version. typescript's standard fixer does not support "Optional chaining" syntax.

You talk about a "typescript standard fixer" and I see within your conf `tslint`, but from what I know `tslint` is [deprecated since 2019](https://medium.com/palantir/tslint-in-2019-1a144c2317a9) in favor of [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint). Is it possible that your issue is due to this ? 🤔
